### PR TITLE
Add recurrence form

### PR DESCRIPTION
**Pull request by dracos**
[Will need to rebuild docker to get new python package]

This adds a recurrence diary-log form - usable by staff or normal users with a complaint on a case, but button only on staff pages in this PR.

Updates the Complaint model to use the new fields needed by the recurrence prototype - note this migration isn't backwards compatible (as it drops the old columns), so you might have to recreate your database if you try out this branch.

Note animation below is out of date, date/time fields are much better now, and there's the whole user thing I've done since thing. I should have done a new animation.

### CODE REVIEW

**Review by zarino — COMMENTED** [review]
When I saw "Search for a person" on `complaint/add/user_search` (as a staff user) it wasn’t clear _who_ it was referring to – complainant or perpetrator. Do you think we could get a title or fieldset legend of "Report as", or "Reporting user" which is how it’s worded on the confirmation screen?

Screenshot 2021-11-19 at 12 20 25

At `complaint/add/user_pick`, do you think we could get some JavaScript that hides the name/email/phone/address inputs until you select "None, details below"?

Screenshot 2021-11-19 at 12 16 42

**Comment by dracos** [comment]
Yes to both things, certainly - yes, I realise I've forgotten the little bits of client JS here and in the reporting branch too - that has a few places where it e.g. needs to hide other box unless "Other" picked and so on.

**Comment by dracos** [comment]
Have pushed a fixup that hopefully implements all that.

**Review by zarino — APPROVED** [review]
:+1:

File: cases/templates/cases/case_detail_staff.html [hunk 1]
```diff
@@ -78,12 +78,13 @@ <h2 class="lbh-heading-h2 section-heading">Case info</h2>
...
 {% if not case.merged_into %}
-
+<div class="nw-button-bar">
     <a href="{% url "case-log-action" case.id %}" class="nw-button">Log an action</a>
-    {% if not duplicate and not case.merged_into %}
-        <a href="{% url "case-merge" case.id %}" class="nw-button nw-button--secondary">Merge into another case</a>
-    {% endif %}
-    <a href="{% url "complaint-add" case.id %}" class="nw-button">Report a reoccurrence</a>
+    <a href="{% url "complaint-add" case.id %}" class="nw-button nw-button--secondary">Report a reoccurrence</a>
+  {% if not duplicate %}
```
Comment (dracos) [inline cases/templates/cases/case_detail_staff.html:84]: any reason this lost the ...and not case.merged_into?

**Pull request merged**